data/source/store/structured/mongo/mongo.go

```diff
@@ -358,30 +350,36 @@ func (s *Session) Destroy(ctx context.Context, id string, condition *request.Con
 	if condition.Revision != nil {
 		query["revision"] = *condition.Revision
 	}
-	changeInfo, err := s.C().RemoveAll(query)
+	changeInfo, err := c.DeleteMany(ctx, query)
 	if err != nil {
 		logger.WithError(err).Error("Unable to destroy data source")
 		return false, errors.Wrap(err, "unable to destroy data source")
 	}
 
 	logger.WithFields(log.Fields{"changeInfo": changeInfo, "duration": time.Since(now) / time.Microsecond}).Debug("Destroy")
-	return changeInfo.Removed > 0, nil
+	return changeInfo.DeletedCount > 0, nil
 }
 
-func (s *Session) get(logger log.Logger, id string, condition *request.Condition) (*dataSource.Source, error) {
+func (c *DataCollection) get(ctx context.Context, logger log.Logger, id string, condition *request.Condition) (*dataSource.Source, error) {
 	results := dataSource.SourceArray{}
 	query := bson.M{
 		"id": id,
 	}
 	if condition != nil && condition.Revision != nil {
 		query["revision"] = *condition.Revision
 	}
-	err := s.C().Find(query).Limit(2).All(&results)
+	opts := options.Find().SetLimit(2)
```

Comment:
Similar to one of my previous comments - wherever we have unique indexes we can simplify the logic and use `FindOne`, because we can't have more than 1 record.